### PR TITLE
APPS-732 Add Side Panel feature

### PR DESCRIPTION
**Pull request by tamarazuk**
This PR introduces the new Side Panel feature to the app template 🚀

React Router is installed here to manage the two different views as it's a very popular framework most React devs are familiar with. 

The views are separated into different components to make it clear what part of the app each component is responsible for. Additionally, moving the `setAppHeight()` handling from `App` to the new `Sidebar` component makes it clear that only the app's Sidebar "area" should communicate the height to Help Scout. This avoids an issue we ran into during development when long side panels were stretching the sidebar app.

This PR also adds a Prettier config to make it a bit easier to work with this repo. 

It should also be noted that an `npm audit fix` was run in this PR as well to address a critical vulnerability. This only updated our package-lock.json file

### CODE REVIEW

**Comment by deivamagalhaes** [comment]
Now that I think about it, I'm not sure if this should be in the app template. IMO the app template is the minimal easy setup to have a React app up and running. If someone does not need a side panel, this adds a lot of unnecessary complexity to the template. Perhaps we could have a different repo with app demos, and this one would go there? 🤔

**Comment by tamarazuk** [comment]
> Perhaps we could have a different repo with app demos, and this one would go there? 🤔

That's a good callout. I think we should discuss this as a team to see what everyone else thinks. I'll start a conversation :)

**Comment by tamarazuk** [comment]
@deivamagalhaes rather than have this discussion block the Side Panel demo, I’ve gone with the branch solution as an interim and have updated my PR with a new base branch. It’s ready for your review. 
We can always change this in the future based on how this discussion goes

File: src/App.tsx [hunk 1]
```diff
@@ -1,47 +1,38 @@
...
-} from "@helpscout/ui-kit";
-import { useEffect, useState } from "react";
+  BrowserRouter as Router,
+  Route,
+  createBrowserRouter,
+  createRoutesFromElements,
+  Outlet,
+  RouterProvider,
+} from 'react-router-dom';
+import { DefaultStyle } from '@helpscout/ui-kit';
```
Comment (deivamagalhaes) [inline src/App.tsx:10]: Should we move relative imports down?

**Review by deivamagalhaes — APPROVED** [review]
LGTM. 

Unrelated: I just noticed a bug with the side panel on the Mailbox if you have the dev tools opened on the side. 😭

File: src/App.tsx (hunk 1 above)
Comment (tamarazuk) [reply src/App.tsx:10]: Nice catch! Should we install `@trivago/prettier-plugin-sort-imports` or just do it manually?

**Comment by deivamagalhaes** [comment]
> LGTM.
> 
> Unrelated: I just noticed a bug with the side panel on the Mailbox if you have the dev tools opened on the side. 😭

Actually not related to dev tools at all, but with the window size: https://i.hlp.sc/Yzwjf6lz 

I'll create a card...

File: src/App.tsx (hunk 1 above)
Comment (tamarazuk) [reply src/App.tsx:10]: I opted not to add any packages and instead manually formatted the imports in e1dc765f12b42c3ab70b88bc2638c0a05c344b3b. I'll go ahead and merge this :)

**Pull request merged**